src/player.cpp

```diff
@@ -180,65 +237,63 @@ PlayerErrors Player::loadWaveform(
 void Player::setWaveformScale(unsigned int soundHash, float newScale)
 {
     auto const &s = std::find_if(sounds.begin(), sounds.end(),
-                                 [&](std::unique_ptr<ActiveSound> const &f)
+                                 [&](std::shared_ptr<ActiveSound> const &f)
                                  { return f->soundHash == soundHash; });
 
     if (s == sounds.end() || s->get()->soundType != TYPE_SYNTH)
         return;
 
-    static_cast<Basicwave*>(s->get()->sound.get())->setScale(newScale);
+    static_cast<Basicwave *>(s->get()->sound.get())->setScale(newScale);
 }
 
 void Player::setWaveformDetune(unsigned int soundHash, float newDetune)
```

Comment:
Aside: This will be controversial but I think we might consider trying to drop support for generating waveforms. I think most game and app development will exclusively use samples. Even music playing apps and sequencers will, at some point, use samples, even if very short ones — unless they go all in with signal processing. But we can't compete with that.

Reply:
My point is that there's a lot of code here that supports a feature that I don't think anyone is using in production. That code needs to be maintained, reviewed, etc.

Reply:
Agree with you! I seen this possibility and sayd myself "cooool I want this"?
Can we think on removing this later? Maybe for version 3? :)

Reply:
Absolutely. I'll file an issue for deprecating this.